src/Apps/InfiniteDiscovery/InfiniteDiscoveryApp.tsx

```diff
@@ -0,0 +1,186 @@
+            </Text>
+
+            <Image
+              src={
+                "https://files.artsy.net/images/diamond-explore-by-flyer01.png"
+              }
+              width="250px"
+              height="250px"
+              lazyLoad
+              alt=""
+            />
+          </Flex>
+          <Flex flexDirection="column">
+            <Box mt={1} p={2}>
+              <Button variant="primaryWhite" onClick={() => setStarted(true)}>
+                Start now
+              </Button>
+            </Box>
+          </Flex>
+        </Flex>
+      )}
+      {started && (
+        <HeroCarousel fullBleed={false} progressbarVariant="dot">
+          {nodes.map(artwork => {
+            return (
+              <DiscoverArtworkContainer key={artwork.id} artwork={artwork} />
+            )
+          })}
+        </HeroCarousel>
+      )}
+    </>
+  )
+}
+
+const DiscoverArtwork = ({ artwork }) => {
+  const { submitMutation: submitSave } = useSaveArtwork()
+
+  const onClick = () => {
+    if (artwork.isSaved) {
+      return submitSave({
+        variables: {
+          input: {
+            artworkID: artwork.internalID,
+            remove: true,
+          },
+        },
+      })
+    }
+
+    return submitSave({
+      variables: {
+        input: {
+          artworkID: artwork.internalID,
+        },
+      },
+    })
+  }
+
+  return (
+    <Flex
+      width="100%"
+      height="100%"
+      flexDirection={["column-reverse", "row"]}
+      alignItems="center"
+      justifyContent="center"
+      textAlign="center"
+      key={artwork.id}
+    >
+      <Box alignItems="center" justifyContent="center" textAlign="center">
+        <>
+          <ResponsiveBox
+            aspectWidth={500}
+            aspectHeight={500}
+            maxHeight={500}
+            maxWidth={500}
+          >
+            <Image
+              width="100%"
+              height="100%"
+              src={artwork.image?.url || undefined}
+            />
+          </ResponsiveBox>
+          <Flex
+            flexDirection="column"
+            alignItems="center"
+            justifyContent="center"
+          >
+            <Flex flexDirection="column" alignItems="flex-start">
+              <Text variant="sm" fontWeight="bold">
+                {artwork.title}
+              </Text>
+              <Text variant={["md", "lg-display"]} color="black60">
+                {artwork.artist?.name}
+                <div style={{ marginLeft: "10px", display: "inline-block" }}>
+                  <SaveButtonBase
+                    artwork={artwork}
+                    isSaved={artwork.isSaved}
+                    onClick={onClick}
+                  />
+                </div>
+              </Text>
+              <Flex alignSelf={"flex-end"}></Flex>
+            </Flex>
+          </Flex>
+        </>
+      </Box>
+    </Flex>
+  )
+}
+
+const DiscoverArtworkContainer = createFragmentContainer(DiscoverArtwork, {
```

Comment:
We should always use relay hooks now. Lots of examples throughout app of `useFragment`.